### ISSUE TITLE
Auto-close tree when tab outside gets focus

### ISSUE THREAD

**Issue by madduck** [piroor/treestyletab#2535]
If I focus a tab outside of a tree, it would be amazing to have an option to automatically collapse the tree.

For example, let's say I am currently looking at the tab called "Child 2":
```
v Parent
    Child 1
  → Child 2
Some other tab
```
If I now click on "Some other tab", then I'd like the tree to collapse by itself:
```
> Parent
→ Some other tab
```
I could not find a combination of options to enable this. Thank you for your consideration!

**Comment by irvinm** [comment]
Have you tried:

![2020-03-27_21-03-02](https://user-images.githubusercontent.com/979729/77811096-821eee00-706e-11ea-8278-7f85f97435ed.jpg)

**Comment by madduck** [comment]
@irvinm yes, I've tried that. This doesn't even do what it says.

**Comment by piroor** [comment]
The option affects only when the newly activated tab has collapsed children. The behavior you expected is not implemented as a builtin.

In my opinion "an activation of a tab with no child" should not trigger the behavior "collapse other tree", instead "an expansion of another tree" should do.

Sorry but I'm negative to introduce new options around this behavior, because too many trivial options will make things complecated and make people confused. Instead it can be done with a helper addon for the purpose. For example:

```javascript
const TST_ID = 'treestyletab@piro.sakura.ne.jp';
browser.tabs.onActivated.addListener(async ({ tabId, windowId }) => {
  const tabs = await browser.runtime.sendMessage(TST_ID, {
    type:   'get-tree',
    window: windowId,
    tabs:   '*'
  });
  const ancestorsOfActive = new Set(tabs.find(tab => tab.id == tabId).ancestorTabIds);
  const rootTabs = tabs.filter(tab => tab.ancestorTabIds.length == 0 && !ancestorsOfActive.has(tab.id));
  browser.runtime.sendMessage(TST_ID, {
    type: 'collapse-tree',
    tabs: rootTabs.map(tab => tab.id)
  });
  browser.runtime.sendMessage(TST_ID, {
    type: 'expand-tree',
    tab:  tabId
  });
});
```

This code snippet is distributed under the public domain, so I hope you or someone create an addon based on it and maintain it.

**Comment by irvinm** [comment]
@piroor if you are not going to work on this, I vote that we close it.
